src/sagemaker/remote_function/core/serialization.py

```diff
@@ -29,6 +28,8 @@
 
 from tblib import pickling_support
 
+# Note: do not use os.path.join for s3 uris, fails on windows
```

Comment:
Wanted to understand more about this shortcoming, if this was researched already?
We may need to add this in [MergeChecklist](https://github.com/aws/sagemaker-python-sdk/blob/master/.github/PULL_REQUEST_TEMPLATE.md) for contributors to be aware.

Reply:
Yes, windows users run across this with s3 related tooling every so often. The `os.sep` on windows is `\\` which doesn't play nice with s3.  Not the first time we've run run in to it, other tools make the same misstep.

When testing the changes to the line endings, we had to change how the s3 uris are constructed to get it making the correct objects on s3 from windows. Python can handle `/` paths for windows filesystems nicely. But s3 will make an object with `\` in the object name instead of a prefix. 

![image](https://github.com/aws/sagemaker-python-sdk/assets/38999128/ca07844f-82be-46ba-bcea-5868e002f803)

The tests were also failing on windows since they expect `/` in the assertions. But were getting `\\`. 

<details>
Before change

```console
pytest ./tests/unit/sagemaker/remote_function/
...

E           AssertionError: expected call not found.
E           Expected: exists('/opt/ml/input/data/sm_rf_user_ws')
E           Actual: exists('/opt/ml/input/data\\sm_rf_user_ws')

venv\Lib\site-packages\mock\mock.py:913: AssertionError
------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------
2023-09-13 18:07:57,722 sagemaker.remote_function INFO     Successfully unpacked workspace archive at '\user\set\workdir'.
2023-09-13 18:07:57,722 sagemaker.remote_function INFO     Did not find any dependency file in workspace directory at '/opt/ml/input/data\sm_rf_user_ws'. Assuming no additional dependencies to install.
================================================================= short test summary info ==================================================================
FAILED tests/unit/sagemaker/remote_function/core/test_serialization.py::test_deserialize_func_deserialization_error - AssertionError: Regex pattern "Error...
FAILED tests/unit/sagemaker/remote_function/core/test_serialization.py::test_deserialize_obj_deserialization_error - AssertionError: Regex pattern "Error ...
FAILED tests/unit/sagemaker/remote_function/core/test_serialization.py::test_serialize_deserialize_service_error - AssertionError: Regex pattern "Failed t...
FAILED tests/unit/sagemaker/remote_function/runtime_environment/test_bootstrap_runtime_environment.py::test_main_success - AssertionError: expected call n...
FAILED tests/unit/sagemaker/remote_function/runtime_environment/test_bootstrap_runtime_environment.py::test_main_channel_folder_does_not_exist - Assertion...
FAILED tests/unit/sagemaker/remote_function/runtime_environment/test_bootstrap_runtime_environment.py::test_main_no_workspace_archive - AssertionError: ex...
FAILED tests/unit/sagemaker/remote_function/runtime_environment/test_bootstrap_runtime_environment.py::test_main_no_dependency_file - AssertionError: expe...
```

</details>

I can add that if you like? Or separate issue/PR to do a search over the sdk for this pattern and correct them?

Reply:
I'l take a note of this to our team and drive the required changes. Thanks for the explanation.